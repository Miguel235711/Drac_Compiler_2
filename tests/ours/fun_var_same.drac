var x;

fun(y){
    x = 3;
}

main(){
    main2 = 3;
    main2();
}
var main;
main(){ -- error, should fail because of redefinition
    var main2,main2; -- 2Â° main2 should fail because of redefinition
}
var main2;
main2(){
    var main2;
}
var main2;-- error, should fail because of redefinition