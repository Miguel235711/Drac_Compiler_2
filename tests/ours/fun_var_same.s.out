.\tests\ours\fun_var_same.drac
type: fun scope: 0 name: printi index: 0
type: fun scope: 0 name: printc index: 1
type: fun scope: 0 name: prints index: 2
type: fun scope: 0 name: println index: 3
type: fun scope: 0 name: readi index: 4
type: fun scope: 0 name: reads index: 5
type: fun scope: 0 name: new index: 6
type: fun scope: 0 name: size index: 7
type: fun scope: 0 name: add index: 8
type: fun scope: 0 name: get index: 9
type: fun scope: 0 name: set index: 10
type: var scope: 1 name: x index: 11
type: fun scope: 0 name: fun index: 12
type: fun scope: 0 name: main index: 14
type: var scope: 1 name: main index: 15
type: var scope: 1 name: main2 index: 17
type: fun scope: 0 name: main2 index: 18
type: var scope: 6 name: y index: 20
type: var scope: 8 name: main2 index: 21
type: var scope: 9 name: main2 index: 22
